src/chrome/client/chromeDebugAdapter/baseCDAState.ts

```diff
@@ -4,13 +4,20 @@ import { CommandText } from '../requests';
 import { IDebugAdapterState } from '../../../debugAdapterInterfaces';
 import { ICommandHandlerDeclarer, CommandHandlerDeclaration, RequestHandlerMappings } from '../../internal/features/components';
 import { injectable } from 'inversify';
+import { ISession } from '../session';
 
 @injectable()
 export abstract class BaseCDAState implements IDebugAdapterState {
     private readonly _requestProcessor: RequestProcessor;
+    protected abstract readonly _session: ISession;
 
-    constructor(requestHandlerDeclarers: ICommandHandlerDeclarer[], requestHandlerMappings: RequestHandlerMappings) {
-        const allDeclarers = requestHandlerDeclarers.concat({ getCommandHandlerDeclarations: () => CommandHandlerDeclaration.fromLiteralObject(requestHandlerMappings) });
+    constructor(
+        requestHandlerDeclarers: ICommandHandlerDeclarer[],
+        requestHandlerMappings: RequestHandlerMappings) {
+        const requestHandlerMappingsWithDefault = { disconnect: () => this.shutdown(), ...requestHandlerMappings };
```

Comment:
I'd add a comment here explaining what you're doing (adding a default disconnect handler). It took me a while to parse out what was actually happening in this code, and at first it looked to me like this would just overwrite whichever 'disconnect' handler got passed.

Reply:
Added comment